runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java

```diff
@@ -336,45 +338,21 @@ private void updateWarning(boolean weaponSwitch)
 			{
 				if (!messageTickGuarded)
 				{
-					for (int i = 1; i <= count; i++)
-					{
-						if (i == 1)
-						{
-							styleBuilder = styleBuilderArray[i];
-							continue;
-						}
-						else if (i == 2 && count != 2)
-						{
-							styleBuilder += (", " + styleBuilderArray[i]);
-							continue;
-						}
-						else if (i == 2 && count == 2)
-						{
-							styleBuilder += (" and " + styleBuilderArray[i]);
-							continue;
-						}
-						else if (i == 3)
-						{
-							styleBuilder += (", and " + styleBuilderArray[i]);
-							continue;
-						}
-					}
-
 					// If the login message has been sent, it will run the corresponding messages if it's a weaponSwitch or just a combatStyle change.
 					if (loginMessageSent)
 					{
 						if (weaponSwitch)
 						{
-							sendChatMessage("This weapon's attack style will grant you " + styleBuilder + " XP.");
+							sendChatMessage("This weapon's attack style will grant XP in: " + builder);
```

Comment:
Maybe these `sendChatMessage` calls should be assignments of a printf-formatted string instead:

```
attackStyleWarningFormat = "This weapon's attack style will grant XP in: %s";
```

and a single `sendChatMessage(String.format(attackStyleWarning, builder))` outside of the `if (loginMessageSent)` if-block.